File: api/src/models/team-invitation.ts
import { z } from 'zod';
import { baseEntitySchema } from './base';

export const invitationStatus = z.enum(['pending', 'accepted', 'rejected']);
export type InvitationStatus = z.infer<typeof invitationStatus>;

export const teamInvitationSchema = baseEntitySchema.extend({
  teamId: z.string(),
  teamName: z.string(),
  invitedPlayerId: z.string().optional(),
  invitedEmail: z.string().email().optional(),
  invitedByPlayerId: z.string(),
  status: invitationStatus.default('pending'),
});

export type TeamInvitation = z.infer<typeof teamInvitationSchema>;

File: PoolDN/PoolDN/Models/TeamInvitation.swift
import Foundation

struct TeamInvitation: Codable, Identifiable {
    let id: String
    let teamId: String
    let teamName: String
    let invitedUserId: String?
    let invitedEmail: String?
    let invitedByUserId: String
    let status: String // "pending", "accepted", "rejected"
    let createdAt: Int
    let updatedAt: Int
}
